README.md

```diff
@@ -8,9 +7,38 @@ This repository provides an OS-agnostic C++ library template with plain [CMake](
   * [Travis](https://travis-ci.org/) + [AppVeyor](https://www.appveyor.com/) script configured
   * dedicated [Doxygen](https://www.stack.nl/~dimitri/doxygen/) target for generating documentation
 
-[![Build Status](https://travis-ci.org/robotology/how-to-export-cpp-library.svg?branch=master)](https://travis-ci.org/robotology/how-to-export-cpp-library) [![Build status](https://ci.appveyor.com/api/projects/status/ak6hx3kp0puo4s0y/branch/master?svg=true)](https://ci.appveyor.com/project/robotology/how-to-export-cpp-library/branch/master) [![Awesome](https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)](https://github.com/onqtam/awesome-cmake)
-
-## Dependencies
+# Overview
+- [💢 Complexities around C++ library](#-complexities-around-c++-library)
+- [🏅 CI and badges](#️-ci-and-badges)
+- [🎛 Dependencies](#-dependencies)
+- [🔨 Build the libraries](#-build-the-libraries)
+- [✂️ Copy and customize this template](#-copy-and-customize-this-template)
+- [🔬 Add a test](#-add-a-test)
+- [🐛 Run the tests](#-run-the-tests)
+- [📝 Generate documentation](#-generate-documentation)
+- [📑 Licensing your library](#-licensing-your-library)
+- [💼 Other template and examples](#-other-template-and-examples)
+
+# 💢 Complexities around C++ library
+This project simplifies the process of taking a bunch of C++ classes/functions and exposing them as a CMake package so that third-party code can use it. However, the risk is that _new users_ underestimate the actual complexity of maintaining a C++ library used by many external users!
+
+A complete and proper training on the art and craft of C++ library maintenance is out of the scope of this project, but we feel that we should at least report some useful link to drive the curiosity and the attention of new users to topics relevant to a proper maintenence of a C++ library.
+
+Problems typically overlooked by new C++ library developers:
+ - [Continuous Delivery](https://www.atlassian.com/continuous-delivery)
+ - [Good practices using Git branching](https://nvie.com/posts/a-successful-git-branching-model/)
+ - [Version handling](https://semver.org/)
+ - [Dynamic loading](https://amir.rachum.com/blog/2016/09/17/shared-libraries/)
+ - [API/ABI backward compatibility](https://abi-laboratory.pro/index.php?view=binary-compatibility)
```

Comment:
I think two additional nice points to add would be: 
* [Symbol Visibility](https://www.akkadia.org/drepper/dsohowto.pdf)
Not sure how, but I think https://github.com/green7ea/cpp-compilation/blob/master/README.md is a super good resource for people that are starting to seriously work with C/C++.

Reply:
I was unsure to add `How To Write Shared Libraries` because it was just too much, but since you are also suggesting it, then I'll just add it 😄 
The other reference is terrific! Added it right away! 👍